KaprekarConstant.cpp

```diff
@@ -0,0 +1,113 @@
+#include <bits/stdc++.h>
+using namespace std;
+
+bool hasSameDigits(const string& number) {
+    int count[10] = {0};
+    for (char digit : number) {
+        if (isdigit(digit)) {
+            int digitValue = digit - '0';
+            count[digitValue]++;
+        }
+    }
+
+    for (int i = 0; i < 10; ++i) {
+        if (count[i] == number.length() && count[i] > 1) {
+            return true;
+        }
+    }
+    return false;
+}
+
+void swapChars(char &firstDigit, char &secondDigit) {
+    char temp = firstDigit;
+    firstDigit = secondDigit;
+    secondDigit = temp;
+}
+
+void descendingOrderSort(string &numbers) {
+    int number = numbers.size();
+    for (int i = 0; i < number - 1; ++i) {
+        for (int j = i + 1; j < number; ++j) {
+            if (numbers[i] < numbers[j]) {
+                swapChars(numbers[i], numbers[j]);
+            }
+        }
+    }
+}
+
```

Comment:
instead of creating a seprate function for sorting in ascending order you can use reverse function